src/test/deploy/functions/validate.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { expect } from "chai";
+import * as fsutils from "../../../fsutils";
+import * as validate from "../../../deploy/functions/validate";
+import * as projectPath from "../../../projectPath";
+import * as FirebaseError from "../../../error";
+import * as sinon from "sinon";
+
+// have to require this because no @types/cjson available
+// tslint:disable-next-line
+const cjson = require("cjson");
+
+describe("validate", () => {
+  describe("functionsDirectoryExists", () => {
+    let sandbox: sinon.SinonSandbox;
+    let resolvePpathStub: sinon.SinonStub;
+    let dirExistsStub: sinon.SinonStub;
+
+    beforeEach(() => {
+      sandbox = sinon.sandbox.create();
```

Comment:
ah - I believe this [`sandbox.create()`](https://sinonjs.org/releases/latest/sandbox/) can just be done above (`const sandbox = sinon.sandbox.create()`). I don't think it has to be done every time.

Reply:
Moved it out to the top, and switched to using the non-deprecated`sinon.createSandbox()`